spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/AdaptCachedBodyGlobalFilter.java

```diff
@@ -70,11 +70,18 @@ public void onApplicationEvent(EnableBodyCachingEvent event) {
 			return chain.filter(exchange);
 		}
 
+		final Mono<String> requestProcessingFinishedFlag = Mono
+				.just("requestProcessingFinishedFlag");
 		return ServerWebExchangeUtils
-				.cacheRequestBody(exchange,
-						(serverHttpRequest) -> chain.filter(
-								exchange.mutate().request(serverHttpRequest).build()))
-				.switchIfEmpty(chain.filter(exchange));
+				.cacheRequestBody(exchange, (serverHttpRequest) -> chain
+						.filter(exchange.mutate().request(serverHttpRequest).build())
+						// suppress empty mono response from FilteringWebHandler, see
+						// https://github.com/spring-cloud/spring-cloud-gateway/pull/1322
```

Comment:
I'd reference https://github.com/spring-cloud/spring-cloud-gateway/issues/1315 here, not the pull requess

Reply:
done